File: DFS/Rock_Paper_Scissors.py
gesture = ['剪刀', '石頭', '布']
visited = [False] * 3
arrangement = [''] * 3


def dfs(layer):
    if layer == 3:
        print(*arrangement, sep='\t\t')
        return

    for i in range(3):
        if visited[i]:
            continue
        visited[i] = True
        arrangement[layer] = gesture[i]
        dfs(layer + 1)
        visited[i] = False

dfs(0)

File: DFS/Rock_Paper_Scissors.cpp
#include <iostream>

using namespace std;

string gesture[3] = {"剪刀", "石頭", "布"};
bool visited[3] = {false};
string arrangement[3];

void dfs(int layer){
    if (layer == 3){
        for (int i = 0; i < 3; i++){
            cout << arrangement[i] << "\t";
        }
        cout << endl;
        return;
    }
    for (int i = 0; i < 3; i++){
        if (visited[i]){
            continue;
        }
        visited[i] = true;
        arrangement[layer] = gesture[i];
        dfs(layer + 1);
        visited[i] = false;
    }
}

int main(){
    dfs(0);

    return 0;
}
